`docker/build.sh` should ignore the cache

### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

Trying to clean build autoware, from newly built docker image (with `./docker/build.sh`)
```
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Found GeographicLib: /usr/lib/x86_64-linux-gnu/libGeographic.so  
-- Found ament_lint_auto: 0.12.4 (/opt/ros/humble/share/ament_lint_auto/cmake)
-- Found OpenCV: /usr (found version "4.5.4") 
CMake Error at CMakeLists.txt:8 (find_package):
  Could not find a package configuration file provided by "magic_enum" with
  any of the following names:

    magic_enumConfig.cmake
    magic_enum-config.cmake

  Add the installation prefix of "magic_enum" to CMAKE_PREFIX_PATH or set
  "magic_enum_DIR" to a directory containing one of the above files.  If
  "magic_enum" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/home/sig/autoware/build/behavior_path_planner/CMakeFiles/CMakeOutput.log".
---
--- stderr: behavior_path_planner
CMake Error at CMakeLists.txt:8 (find_package):
  Could not find a package configuration file provided by "magic_enum" with
  any of the following names:

    magic_enumConfig.cmake
    magic_enum-config.cmake

  Add the installation prefix of "magic_enum" to CMAKE_PREFIX_PATH or set
  "magic_enum_DIR" to a directory containing one of the above files.  If
  "magic_enum" provides a separate development package or SDK, be sure it has
  been installed.


---
Failed   <<< behavior_path_planner [39.2s, exited with code 1]
Aborted  <<< pure_pursuit [3min 10s]                      
Aborted  <<< pid_longitudinal_controller [4min 27s]
Aborted  <<< obstacle_avoidance_planner [4min 41s]        
Aborted  <<< motion_velocity_smoother [5min 39s]          
Aborted  <<< mission_planner [6min 1s]                    
Aborted  <<< mpc_lateral_controller [6min 21s]            
Aborted  <<< pointcloud_preprocessor [18min 21s]                                             

Summary: 93 packages finished [29min 58s]
  1 package failed: behavior_path_planner
  7 packages aborted: mission_planner motion_velocity_smoother mpc_lateral_controller obstacle_avoidance_planner pid_longitudinal_controller pointcloud_preprocessor pure_pursuit
  11 packages had stderr output: autoware_cmake autoware_lint_common behavior_path_planner common_sensor_launch lidar_apollo_instance_segmentation map_tf_generator ndt_omp tier4_auto_msgs_converter tvm_utility velodyne_description velodyne_pointcloud
  134 packages not processed
```

The new dependency `magic_enum` has not been installed in the docker image. It should normally be built by `rosdep` here:
https://github.com/autowarefoundation/autoware/blob/d34355c8d5ce26405546c183405f6fa0f76e001e/docker/autoware-universe/Dockerfile#L34

The problem seems that the `./docker/build.sh` script does not ignore the cache. It can be confirmed from the output of `./docker/build.sh`:
```
...

#10 [prebuilt devel  8/13] RUN --mount=type=ssh   ./setup-dev-env.sh -y --no-cuda-drivers universe   && pip uninstall -y ansible ansible-core   && mkdir src   && vcs import src < autoware.repos   && rosdep update   && DEBIAN_FRONTEND=noninteractive rosdep install -y --ignore-src --from-paths src --rosdistro "humble"   && apt-get clean   && rm -rf /var/lib/apt/lists/*
#10 CACHED

...
```

Adding `--no-cache` in `./docker/build.sh` seems to do the trick:
```
...

#15 [devel devel  8/13] RUN --mount=type=ssh   ./setup-dev-env.sh -y --no-cuda-drivers universe   && pip uninstall -y ansible ansible-core   && mkdir src   && vcs import src < autoware.repos   && rosdep update   && DEBIAN_FRONTEND=noninteractive rosdep install -y --ignore-src --from-paths src --rosdistro "humble"   && apt-get clean   && rm -rf /var/lib/apt/lists/*
#15 1.409 Run the setup in non-interactive mode.
#15 2.079 Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
#15 2.079 Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
#15 3.141 Get:3 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [795 kB]
#15 3.383 Get:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]
#15 3.687 Get:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]
#15 3.982 Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
#15 4.146 Get:7 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [681 kB]
#15 4.281 Get:8 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [745 kB]
#15 4.485 Get:9 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [4732 B]
#15 4.697 Get:10 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
#15 4.725 Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
#15 6.253 Get:12 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
#15 6.260 Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [8978 B]
#15 6.260 Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [730 kB]
#15 6.279 Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1077 kB]
#15 6.339 Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1007 kB]
#15 6.434 Get:17 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [3520 B]
#15 6.434 Get:18 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [7286 B]
#15 6.965 Fetched 25.3 MB in 6s (4577 kB/s)
#15 6.965 Reading package lists...
#15 7.458 Reading package lists...
#15 7.935 Building dependency tree...
#15 8.036 Reading state information...
#15 8.141 The following NEW packages will be installed:
#15 8.141   sudo
...
```



### Expected behavior

up to date docker image image

### Actual behavior

docker image is built using outdated cache

### Steps to reproduce

N/A

### Versions

_No response_

### Possible causes

_No response_

### Additional context

_No response_

--- a/docker/build.sh
+++ b/docker/build.sh
@@ -52,7 +52,7 @@ fi
 export BUILDKIT_STEP_LOG_MAX_SIZE=10000000
 
 set -x
-docker buildx bake --load --progress=plain -f "$SCRIPT_DIR/autoware-universe/docker-bake.hcl" \
+docker buildx bake --no-cache --load --progress=plain -f "$SCRIPT_DIR/autoware-universe/docker-bake.hcl" \
     --set "*.context=$WORKSPACE_ROOT" \
     --set "*.ssh=default" \
     --set "*.platform=$platform" \

